builtin/logical/mysql/path_role_create.go

```diff
@@ -51,9 +52,15 @@ func (b *backend) pathRoleCreateRead(
 	}
 
 	// Generate our username and password. MySQL limits user to 16 characters
-	displayName := req.DisplayName
-	if len(displayName) > 10 {
-		displayName = displayName[:10]
+	displayName := name
```

Comment:
okay, if this is changed, then #1605 would be resolved - personally i like to have the full display name as part of the user, as it tells me what source the user was created from - a token / root key and so forth - changing this will make mysql adapter inconsistent with all other secret backends - so it would probably be atomic fix to either do at all backends, or nowhere imo